synapse/dendrite/parser.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import synapse.telepath as s_telepath
+import synapse.lib.service as s_service
+
+logger = logging.getLogger(__name__)
+
+class Parser:
+    '''
+    Parser provides a base implementation for processing parsing jobs in Dendrite.
+
+    The Parser is intended to be used by sub-classes for the handling of basic operations required for processing jobs
+    in Dendrite. Sub-classes must implement ``register`` to register a MIME type of interest with a specific queue
+    name. Some care should be taken to ensure that queue names do not collide on a given deployment.
+
```

Comment:
The more i think about it, the more it seems to me that these "queue names" should all just be GUIDs of some kind or another.  Otherwise, we may end up complicating the role of eventual "orchestration" which would spin up and spin down workers.